README.md

```diff
@@ -107,23 +110,65 @@ return fmt.Errorf("invalid port: %v", err)
 
 ## Apply the patch
 
-To apply the patch, `cd` to your Go project's directory.
-
-```shell
-$ cd ~/go/src/example.com/myproject
-```
-
-Run `gopatch` on the project, supplying the previously written patch with the
-`-p` flag.
-
-```shell
-$ gopatch -p ~/s1028.patch ./...
-```
-
-This will apply the patch on all Go code in your project.
-
-Check if there were any instances of this issue in your code by running
-`git diff`.
+- To apply the patch, `cd` to your Go project's directory.
+
+  ```shell
+  $ cd ~/go/src/example.com/myproject
+  ```
+
+  Run `gopatch` on the project, supplying the previously written patch with the
+  `-p` flag.
+
+  ```shell
+  $ gopatch -p ~/s1028.patch ./...
+  ```
+
+  This will apply the patch on all Go code in your project.
+
+  Check if there were any instances of this issue in your code by running
+  `git diff`.
+- To list the matches instead, `cd` to your Go project's directory.
+
+  ```shell
+  $ cd ~/go/src/example.com/myproject
+  ```
+
+  Run `gopatch` on the project, supplying the previously written patch with the
+  `-p` flag along with '-l' flag.
+
+  ```shell
+  $ gopatch -l -p ~/s1028.patch ./...
+  ```
+
+  This will turn on lint mode and will show you matches instead of modifying all
+  the Go code in your project. To provide more context on what the patch does, if
+  there were description comments in the patch, they will also get displayed at 
+  the top. To learn more about description comments jump to section [here](#description-comments)
+  
+  For example if we applied patch ~/s1028 to our testfile error.go
+  ```shell
+  $ gopatch -l -p ~/s1028.patch ./testdata/test_files/lint_example/
+  ```
+  Output would be : 
+  ```
+  This patch replaces instances of fmt.Sprintf()
+  with fmt.Errorf()
+  
+  Patch files can be applied to mutiple files
+  --- gopatch/testdata/test_files/lint_example/error.go
+  +++ changed File
```

Comment:
The `+++ changed File` is not informative. Could we change it to the:
```
+++ gopatch/testdata/test_files/lint_example/error.go
```

Reply:
Got it! Made all these changes